event.go

```diff
@@ -859,16 +855,72 @@ func (e *Event) Membership() (string, error) {
 	default:
 		panic(e.invalidFieldType())
 	}
-	if fields.Type != MRoomMember {
-		return "", fmt.Errorf("gomatrixserverlib: not an m.room.member event")
+	if fields.Type != eventType {
+		return fmt.Errorf("gomatrixserverlib: not a %s event", eventType)
 	}
+	return json.Unmarshal(fields.Content, &content)
```

Comment:
Should we check that `len(fields.Content) > 0`?

Reply:
No, as it will just give an unmarshal error in that case.